File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        D = dict()
        for i in nums:
            D[i]=D.get(i,0)+1

        arr=[]
        for num,freq in D.items():
            arr.append([freq,num])
        arr.sort()

        res=[]
        while len(res)<k:
            res.append(arr.pop()[1])
        return res

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0, mid=0;
        int h = nums.size()-1;
        for(int i=0; i<nums.size(); i++){
            mid = l+(h-l)/2; //l=[0]=3; mid=[2]=5; h=[5]=2;
            cout<<"beg"<<nums[l]<<" "<<nums[mid]<<" "<<nums[h]<<endl; 
            if(nums[mid]>nums[h]){l=mid;}
            else{if(nums[mid]<nums[l]){h=mid;}
            else{return nums[l];}}
            cout<<nums[l]<<" "<<nums[mid]<<" "<<nums[h]<<endl;
        }
        return nums[h];
    }
};
